vispy/visuals/shaders/compiler.py

```diff
@@ -36,7 +37,7 @@ class Compiler(object):
     def __init__(self, **shaders):
         # cache of compilation results for each function and variable
         self._object_names = {}  # {object: name}
-        self.shaders = shaders
+        self.shaders = weakref.WeakValueDictionary(**shaders)
```

Comment:
It surprises me that this is necessary. It also feels a bit strange, since the vert and frag shaders feel "tightly bound". @campagnola WDYT?